gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/http/connector/AbstractConnector.java

```diff
@@ -106,27 +106,45 @@ public AbstractConnector(T endpoint) {
     public void request(ProxyRequest proxyRequest, Handler<ProxyConnection> proxyConnectionHandler) {
         // For Vertx HTTP client query parameters have to be passed along the URI
         final String uri = appendQueryParameters(proxyRequest.uri(), proxyRequest.parameters());
-
-        // Add the endpoint reference in metrics to know which endpoint has been invoked while serving the request
-        proxyRequest.metrics().setEndpoint(uri);
+        proxyRequest.headers().remove(HttpHeaders.HOST);
+
+        String customHost = null;
+
+        // Enhance proxy request with endpoint configuration
+        if (endpoint.getHeaders() != null && !endpoint.getHeaders().isEmpty()) {
+            for (String httpHeaderName : endpoint.getHeaders().keySet()) {
+                String httpHeaderValue = endpoint.getHeaders().get(httpHeaderName);
+                if (httpHeaderName.equals(HttpHeaders.HOST)) {
+                    customHost = httpHeaderValue;
+                } else {
+                    proxyRequest.headers().set(httpHeaderValue, httpHeaderValue);
```

Comment:
Like discussed here is a little mistake. Maybe we can add a small test on it 
```suggestion
                    proxyRequest.headers().set(httpHeaderName, httpHeaderValue);
```